### PR TITLE
Implemented testing of home page, details page and search bar

### PR DESCRIPTION
**Pull request by LeonardoMolleker**
Screenshot_1
Screenshot_2
Screenshot_3
Screenshot_4

### CODE REVIEW

File: lib/pages/home_body_content.dart [hunk 1]
```diff
@@ -46,32 +45,35 @@ class HomeBodyContent extends StatelessWidget {
   }
 
   List<Widget> buildList(BuildContext context, AsyncSnapshot snapshot) {
```
Comment (MarcosCGlobant) [inline lib/pages/home_body_content.dart]: can you abstract this into a new widget file?

File: lib/pages/home_page.dart [hunk 2]
```diff
@@ -14,6 +14,18 @@ class Home extends StatelessWidget {
...
   }) : super(key: key);
 
+  @override
+  _HomeState createState() => _HomeState();
+}
+
+class _HomeState extends State<Home> {
+  @override
+  void initState() {
+    // TODO: implement initState
```
Comment (MarcosCGlobant) [inline lib/pages/home_page.dart]: missed a comment here

File: test/build_widget.dart [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+Widget buildWidget(Widget child) {
+  return MaterialApp(
+    debugShowCheckedModeBanner: false,
+    title: 'Flutter Demo',
+    theme: ThemeData(
+      primarySwatch: Colors.blue,
+    ),
+    home: child,
+  );
+}
```
Comment (MarcosCGlobant) [inline test/build_widget.dart]: add an extra line at the end, Ctrl + Alt + L should fix it

File: test/movie_details_test.dart [hunk 4]
```diff
@@ -0,0 +1,97 @@
+import 'package:cached_network_image/cached_network_image.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:tmdb_movies/utils/testing_constants.dart';
+import '../lib/models/movie_result.dart';
+import 'package:flutter/material.dart';
+import '../lib/pages/movie_details_page.dart';
+import '../lib/utils/string_constants.dart';
+import 'build_widget.dart';
+
+
```
Comment (MarcosCGlobant) [inline test/movie_details_test.dart]: remove extra line

Comment (LeonardoMolleker) [reply test/movie_details_test.dart]: done

File: test/build_widget.dart (hunk 3 above)
Comment (LeonardoMolleker) [reply test/build_widget.dart]: done

File: lib/pages/home_page.dart (hunk 2 above)
Comment (LeonardoMolleker) [reply lib/pages/home_page.dart]: removed

File: lib/pages/home_body_content.dart (hunk 1 above)
Comment (LeonardoMolleker) [reply lib/pages/home_body_content.dart]: moved

**Review by MarcosCGlobant — APPROVED** [review]
LGTM

**Review by FranciscoGaraventa — APPROVED** [review]
LGTM

**Pull request merged**